pkgs/development/compilers/circt/default.nix

```diff
@@ -6,22 +6,30 @@
 , git
 , fetchFromGitHub
 , ninja
+, substituteAll
 }:
 
 let
   pythonEnv = python3.withPackages (ps: [ ps.psutil ]);
 in
 stdenv.mkDerivation rec {
   pname = "circt";
-  version = "1.45.0";
+  version = "1.48.0";
   src = fetchFromGitHub {
     owner = "llvm";
     repo = "circt";
     rev = "firtool-${version}";
-    sha256 = "sha256-yzXYiqRIwV3bkMfvmduow3QWJASXhOspM8CHZPN2/uE=";
+    sha256 = "sha256-8mqh3PPfB50ZkiJ+1OjclWw19t6OLv1mNiVkBnDz5jQ=";
     fetchSubmodules = true;
   };
 
+  patches = [
+    # circt uses git to check its version,
+    # but when cloned on nix it can't access git.
+    # So this patch hard codes the version.
+    (substituteAll { src = ./circt-version.diff; version = src.rev; })
+  ];
```

Comment:
Can we just do `substituteInPlace cmake/modules/GenVersionFile.cmake --replace "unknown git version" "$version"`

Reply:
Thanks for the suggestion!
Yes, I think this would work too.

Reply:
I've updated the code.
Still building on my computer to check if everything works.